### ISSUE TITLE
How to include MMWormhole in swift

### ISSUE THREAD

**Issue by etiennea** [leecrossley/cordova-plugin-apple-watch#16]
Anyone had any luck with creating a bridge file successfully for MMWormhole could shed a tiny bit of light? Would be very much appreciated!

**Comment by leecrossley** [comment]
Which part are you stuck with? Please amend the issue with a specific question or problem you are stuck with, thanks.

**Comment by etiennea** [comment]
// assumes your WatchKit extension references the packaged MMWormhole library `libmmwormhole.a`

let wormhole = MMWormhole(applicationGroupIdentifier: "group.com.yourcompany", optionalDirectory: nil)

This line does not let me compile... I think it is because I cannot figure out how the reference the libmmwormhole.a? Does the 'cordova plugin add xxx' do this for me?

I cannot do #import MMWormhole.h in swift right? So I ready somewhere you had to add a bridge header I did that but I still cannot get it to compile must be missing something...

**Comment by leecrossley** [comment]
`cordova plugin add cordova-plugin-apple-watch` will add the reference to the libmmwormhole.a library for the plugin used in your cordova app but you may need to add the reference to your WatchKit extension.

You will also need to do `#import "MMWormhole.h"` in Objective C or `import MMWormhole` in Swift in your WatchKit extension.

**Comment by etiennea** [comment]
the `import MMWormhole` line in swift still gives me the error `no such module mmwormhole` and I have already spent 6h trying to figure it out. I can't imagine that I am the only one who will have this issue.

**Comment by leecrossley** [comment]
You need to add the reference to `libmmwormhole.a` in your WatchKit Extension, the "plugin add" can't do this part.

**Comment by leecrossley** [comment]
If you're really stuck, you can just drag MMWormhole.h and libmmwormhole.a from the filesystem (plugin directory or wherever else you have it) into your WatchKit Extension project.

**Comment by etiennea** [comment]
I have tried both, when on the target I have linked the binary, to no effect. 
I have also dragged these and added to the target has no impact.

**Comment by leecrossley** [comment]
There seems to be good advice for including libraries [here](http://stackoverflow.com/questions/24057756/import-framework-in-swift-project-xcode-6-0-beta) - I guess you may have ready read this.

Failing that, add your own question to SO, as this is a generic problem. I need to find time to clean up my demo projects and push them to Github.

**Comment by etiennea** [comment]
Ok so I managed there are a lot of steps!! 
I will do a pull request with some extra info in the readme if you want.

**Comment by leecrossley** [comment]
Sound good, I'm glad you got there :)

**Comment by devnill** [comment]
After a spending more time than I'm proud to admit I figured out how to get this to work. Heres the steps I took to get it to work:
- In the extensions InterfaceController.swift, I added `let wormhole = MMWormhole(applicationGroupIdentifier: "group.com.groupnamehere", optionalDirectory: nil)`
- Under my extension, I created a Appname-Bridging-Header.h file. Between the #define and #endif, I added #import "MMWormhole.h"
- I dragged MMWomhole.h from the plugin directory into my extension
- In the extension's build settings I added the the bridging-header to the "Objective-C Bridging Header" section. 
- In the extension's build phases I added libmmwormhole.a under "Link Binary With Libraries"

I _think_ that is all. Let me know if I'm missing a step.
